src/esp/physics/RigidBase.h

```diff
@@ -89,14 +89,20 @@ class RigidBase : public Magnum::SceneGraph::AbstractFeature3D {
   }
   /**
    * @brief Initializes the @ref RigidObject or @ref RigidScene that inherits
-   * from this class
-   * @param physicsAttributes The template structure defining relevant
+   * from this class.  This is overridden
+   * @param resMgr a reference to ResourceManager object
+   * @param handle The handle for the template structure defining relevant
    * phyiscal parameters for this object
    * @return true if initialized successfully, false otherwise.
    */
-  virtual bool initialize(
-      const assets::ResourceManager& resMgr,
-      const assets::AbstractPhysicsAttributes::ptr physicsAttributes) = 0;
+  virtual bool initialize(const assets::ResourceManager& resMgr,
+                          const std::string& handle) = 0;
+
+  /**
+   * @brief Finalize the creation of @ref RigidObject or @ref RigidScene that
+   * inherits from this class.
+   */
+  virtual void finalizeObject() = 0;
```

Comment:
Are we assuming that nothing in `finalizeObject` or `finalizeObject_LibSpecific` can fail? Should this be a boolean return with destruction/de-allocation guard in `PhysicsManager`?

Reply:
good idea.  I'll update it to support that.